sphinxcontrib/spelling/role.py

```diff
@@ -1,17 +1,14 @@
-from docutils.parsers.rst import roles
+from docutils import nodes
 
 from . import directive
 
 
-class WordRole(roles.GenericRole):
-    """Let the user indicate that inline text is spelled correctly.
-    """
-
-    def __call__(self, role, rawtext, text, lineno, inliner,
-                 options={}, content=[]):
-        env = inliner.document.settings.env
-        docname = env.docname
-        good_words = text.split()
-        directive.add_good_words_to_document(env, docname, good_words)
-        return super().__call__(role, rawtext, text, lineno, inliner,
-                                options=options, content=content)
+def spelling_word(role, rawtext, text, lineno, inliner,
+                  options={}, content=[]):
+    """Let the user indicate that inline text is spelled correctly."""
+    env = inliner.document.settings.env
+    docname = env.docname
+    good_words = text.split()
+    directive.add_good_words_to_document(env, docname, good_words)
+    node = nodes.Text(text)
```

Comment:
This does look much simpler. Is the real fix that this Text node is initialized differently than when GenericRole creates it?

Reply:
Yes, I think so. I struggled to understand the route that the original code took to create the final Text node, but it does seem like that is the case.